backend/dataall/core/context.py

```diff
@@ -0,0 +1,42 @@
+"""
+API for request context.
+Request context is a storage for associated with the request and should accessible from any part of application
+that in the request scope
+
+The class uses Flask's approach to handle request: ThreadLocal
+That approach should work fine for AWS Lambdas and local server that uses Flask app
+"""
+
+from dataclasses import dataclass
+from typing import List
+
+from dataall.db.connection import Engine
+from threading import local
+import opensearchpy
+
+
+_request_storage = local()
```

Comment:
[Clarification] - it creates a local instance for this thread. In this local instance we can store info about the context of this particular API call keeping it isolated from other API calls